src/store/index.js

```diff
@@ -27,29 +21,23 @@ export default new Vuex.Store({
     setSearchQuery: (state, value) => {
       state.searchQuery = value
     },
-    setFilteredMovies: (state, value) => {
-      state.filteredMovies = value
+    setMovies: (state, value) => {
+      state.movies = value
     }
   },
   actions: {
-    updateFilteredMovies: ({commit, state}) => {
-      const filteredMovies = state.movies.filter(movie => {
-        if (state.searchField === 'genre') {
-          return movie.genres.includes(state.searchQuery)
-        }
-        if (state.searchField === 'title') {
-          return movie.title.includes(state.searchQuery) || movie.overview.includes(state.searchQuery)
-        }
+    loadMovies: async ({commit}) => {
+      let newVar = await movieApi.getMovies();
+      commit('setMovies', newVar)
+    },
+    updateFilteredMovies: async ({commit, state}) => {
+      const filteredMovies = await movieApi.getMovies({
+        sortBy: state.sort,
+        searchBy: state.searchField,
+        search: state.searchQuery,
+        sortOrder: 'desc'
       })
-        .sort((a, b) => {
-          if (state.sort === 'date') {
-            return new Date(a.release_date) > new Date(b.release_date) ? -1 : 1
-          }
-          if (state.sort === 'rating') {
-            return a.vote_average > b.vote_average ? -1 : 1
-          }
-        })
-      commit('setFilteredMovies', filteredMovies)
+      commit('setMovies', filteredMovies)
```

Comment:
A best practice would be storing action specific names in a "actionTypes" file and using it to access the action name or defining them as dynamic properties.
E.g. [actionTypes.SET_MOVIES](context, query) {...} in movies store module.
commit((actionTypes.SET_MOVIES, ...)
Using this approach you avoid accidental typos in actions across the app.